src/misc/rwqueue.cpp

```diff
@@ -102,6 +102,13 @@ bool RWQueue<T>::IsEmpty()
 	return queue.empty();
 }
 
+template <typename T>
+bool RWQueue<T>::IsFull()
+{
+	std::lock_guard<std::mutex> lock(mutex);
+	return (queue.size() >= capacity);
+}
+
```

Comment:
I'm confused. Why do we need `IsFull()` at all? All we need is to use `!IsEmpty()` in the client code.

Reply:
They're not the same though (`!Is empty()` means you might have 1 up to the max)

Reply:
Ok... brain malfunction, carry on 😆

Reply:
Np!